apps/system/js/accessibility.js

```diff
@@ -309,12 +337,17 @@
           return;
       }
 
-      this.speak(aDetails.data, null, {
+      this.speak(aDetails.data, function hintsCallback() {
+        if (options.hints) {
+          this.setHintsTimeout(aDetails, options.hints);
```

Comment:
Had contemplated going this route (in fact it's what I started with), but yeah, it does seem to be a better solution than the chained speak calls.  Will do this instead.